lib/jxl/dec_frame.cc

```diff
@@ -909,6 +909,12 @@ Status FrameDecoder::FinalizeFrame() {
         "FinalizeFrame called before the frame was fully decoded");
   }
 
+  if (!finalized_dc_) {
+    JXL_DASSERT(allow_partial_frames_);
```

Comment:
Are we sure that this DASSERT cannot be triggered by erraneous usage?

Reply:
Probably best to use regular JXL_ASSERT in cases like this where it's not in a hot loop — makes it more likely to catch things.

Reply:
Yup, will change to DASSERT.
decoded_ac_global_ checked in preceding `if` is set to true at the end of ProcessACGlobal that is invoked if finalized_dc_. This assertion is more for readability than for safety.

Reply:
PTAL